### ISSUE TITLE
Image pushes to Quay from Travis are flaky

### ISSUE THREAD

**Issue by ericchiang** [coreos/dex#489]
We're getting some weird errors from Travis right now when pushing to Quay. For example see https://travis-ci.org/coreos/dex/jobs/140677161#L297

The push seems to succeed but Travis still exists with exit status 1.

```
The command "DEX_TEST_DSN="sqlite3://:memory:" ./test-functional" exited with 0.
dpl.0
2.05sWarning! PATH is not properly set up, '/usr/local/rvm/gems/ruby-2.2.3/bin' is not at first place,
         usually this is caused by shell initialization files - check them for 'PATH=...' entries,
         it might also help to re-add RVM to your dotfiles: 'rvm get stable --auto-dotfiles',
         to fix temporarily in this shell session run: 'rvm use ruby-2.2.3'.
Warning, new version of rvm available '1.27.0', you are using older version '1.26.11'.
You can disable this warning with:    echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable  auto-update  with:    echo rvm_autoupdate_flag=2 >> ~/.rvmrc
Fetching: dpl-1.8.17.gem (100%)
Successfully installed dpl-1.8.17
1 gem installed
Warning! PATH is not properly set up, '/usr/local/rvm/gems/ruby-2.2.3/bin' is not at first place,
         usually this is caused by shell initialization files - check them for 'PATH=...' entries,
         it might also help to re-add RVM to your dotfiles: 'rvm get stable --auto-dotfiles',
         to fix temporarily in this shell session run: 'rvm use ruby-2.2.3'.
dpl.1
Installing deploy dependencies
!!! Script support is experimental !!!
dpl.2
Preparing deploy
dpl.3
Deploying application
----> build quay.io/coreos/dex:v0.5.0
Sending build context to Docker daemon 139.4 MB
Step 0 : FROM quay.io/brianredbeard/corebox
latest: Pulling from brianredbeard/corebox
Status: Downloaded newer image for quay.io/brianredbeard/corebox:latest
 ---> 8ef8f25ba499
Step 1 : ADD bin/dex-worker /opt/dex/bin/dex-worker
 ---> cf850e9cba95
Removing intermediate container 71114e263601
Step 2 : ADD bin/dex-overlord /opt/dex/bin/dex-overlord
 ---> dd97a8f14fa3
Removing intermediate container 697e20c9224c
Step 3 : ADD bin/dexctl /opt/dex/bin/dexctl
 ---> ba111715c4e4
Removing intermediate container d40b1d12e3f3
Step 4 : ENV DEX_WORKER_HTML_ASSETS /opt/dex/html/
 ---> Running in f517ccc1b7ae
 ---> 00c8707ff43e
Removing intermediate container f517ccc1b7ae
Step 5 : ADD static/html/* $DEX_WORKER_HTML_ASSETS
 ---> 77bf5fb821cf
Removing intermediate container 01cac48e74a2
Step 6 : ENV DEX_WORKER_EMAIL_TEMPLATES /opt/dex/email/
 ---> Running in 645f83d3a40e
 ---> 462cbfcd2a78
Removing intermediate container 645f83d3a40e
Step 7 : ADD static/email/* $DEX_WORKER_EMAIL_TEMPLATES
 ---> 6eb960a495c4
Removing intermediate container 3126cb4f49da
Step 8 : ADD static/fixtures/emailer.json.sample $DEX_WORKER_EMAIL_TEMPLATES/emailer.json
 ---> 643e73b17d97
Removing intermediate container 54f2924c6d42
Successfully built 643e73b17d97
----> push image quay.io/coreos/dex:v0.5.0
logging in as coreos+jenkins
WARNING: login credentials saved in /home/travis/.docker/config.json
Login Succeeded
The push refers to a repository [quay.io/coreos/dex] (len: 1)
The push refers to a repository [quay.io/coreos/dex] (len: 1)
Script failed with status 1
failed to deploy
```

**Comment by ericchiang** [comment]
This is actively breaking the build. Making a p0

**Comment by ericchiang** [comment]
Fixed by #493
